### DISCUSSION TITLE
Lifebars: new palfx variation

### DISCUSSION THREAD

**Discussion by MUGEN-Slayer** [ikemen-engine/Ikemen-GO#1350] [Feature Request]
### Is your feature request related to a problem? Please describe.

Let's say I want to make the front layer of the health bar turn red and return to the base color, I tried with sinadd:
![sinebar](https://github.com/ikemen-engine/Ikemen-GO/assets/68881657/e40a76dc-a398-4f03-bcae-648bf0df8b3c)

It didn't work because of the sine motion, obviously. The front layer goes from base to red, back and turns blue.
So why try "sine-wave"? Because currently it's the only palfx that can be used to emulate an animation, and in some specific cases it even works to do what I've tried here, keyword being "specific".

So it would be great if we could have a variation of palfx that just does "half wave". With such a variation, the attempt here would be possible:
![halfsine](https://github.com/ikemen-engine/Ikemen-GO/assets/68881657/5d5b3e20-c5f9-42f2-a914-a101bf4c8cd4)

Obviously I could have used an animation for this, but quoting the wiki:
"Hint: When it comes to lifebars the main use case of palfx is adjusting palette without need to duplicate sprites to do so."

**Comment by SuperFromND** [comment]
Either this, or some way to somehow have a PalFx use an `abs()` function to make it a half-wave would be nice to have.

For context: `abs()` is a common math function in a lot of languages (MUGEN CNS included!) that returns the absolute value of its input (e.g. `abs(-1)` = 1, `abs(1)` = 1, etc.).  When applied to a sine wave, it turns it into a half-sine:

![Sine vs Half-Sine](https://github.com/ikemen-engine/Ikemen-GO/assets/22881403/f62dc5b2-132b-48d6-a0d2-12cced21a5ac)

As for how to write that out, maybe a new parameter for using absolutes could work:
```ini
p1.front.palfx.sinadd= 255,-50,0,16
p1.front.palfx.sinabs= 1 ; new!
```
Where a value of 1 uses absolutes, a value of 0 doesn't use any absolute at all, and maybe -1 for giving inverted absolute values.

**Comment by potsmugen** [comment]
I missed this earlier, but for a halfwave-ish effect you can use the add parameter to start the wave from the middle point.

For instance:
```
p1.front.palfx.add= 128, 0, 0
p1.front.palfx.sinadd= 128, 0, 0, 60
```
Will never subtract red.

Chars are like this too, so adding this parameter seems a bit overkill. That said it wouldn't hurt either.

**Reply by MUGEN-Slayer** [reply]
Oh! Good to know. I only advocate for features that bring something new to the table, if there is already a way to do that then such variation is pointless (unless there was an edge case).
This ended up making me test with 32-bit sprites and it works with them too, the last time (years ago) I tried to apply palfx to these sprites  it didn't work so this is an extra. ~~Unfortunately you still can't apply RGB to fonts with 32-bit sprites, but I digress.~~
